packages/cli/src/WorkflowRunner.ts

```diff
@@ -741,6 +741,12 @@ export class WorkflowRunner {
 					childExecutionIds.splice(executionIdIndex, 1);
 				}
 
+				if (message.data.result === undefined) {
+					await Db.collections.Execution.update(message.data.executionId, {
```

Comment:
Instead of saving data directly as done here, we should actually call `this.processError` and add the desired behavior there, such as changing the status to `failed`.  That function also ensures that we call the correct `workflowExecuteAfter` hooks, that include saving more data if necessary and doing all cleanup.